Hottahd/issue49

hottahd opened the pull request:
3次元の可視化方法を検討しました。

最も簡単な3次元問題ということでケルビンヘルムホルツ不安定の3次元問題を設定(これが必要かは検討config.yamlを変えているだけ)しました。

pyMISOにVTKファイルを出力する関数を追加。

hottahd committed:
堀田が使っているminiforgeではデフォルトのpythonが3.10なので、最低要件は3.10とした

hottahd committed:
data.laodで読み込んだデータのn_outputを記録することにした。

hottahd committed:
座標の読み込みはマージンを含めないこととした。data.loadで読み込むデータではマージンを含めていない

hottahd committed:
pyMISOのデータをvtkに出力する関数を追加

hottahd committed:
3D KH問題を追加

hottahd committed:
ruffのエラーに対応

hottahd committed:
copilotの提案に従って改訂

iijimahr commented on pyMISO/write.py:
```diff
@@ -0,0 +1,40 @@
+import pyvista as pv
```
`project.optional-dependencies` に `pyvista` の追加をお願いします。
3Dを必須としないなら、`vis`とは別に`vis3d`を作っても良いかも。。

iijimahr commented on pyMISO/write.py:
```diff
@@ -0,0 +1,45 @@
+import pyMISO
+
+
+def write_scalar_vtk(data, n_output, var, output_path):
```
私も関数内でのimportで良い気がします。
または、以下のようなエラーハンドリングで3Dライブラリがなくても動くようにしてはどうでしょうか。

```python
# 出典: https://github.com/sunpy/sunkit-magex/blob/main/sunkit_magex/pfss/pfss.py
HAS_NUMBA = False
try:
    import numba
    HAS_NUMBA = True
except Exception:
    pass
```

iijimahr commented on pyMISO/__init__.py:
`write_scalar_vtk` だけ出ているのがモジュール構造として少し違和感があります。
`io.write_scalar_vtk` や `writers.write_scalar_vtk` 、 `utils.write_scalar_vtk` などはどうでしょうか。
`Data.write_scalar_vtk` も自然な気がします。

こちらは好みの問題という気もするので採否はお任せです。

hottahd replied:
ありがとうございます。全部対応したつもりです。

iijimahr reviewed (changes requested):
ありがとうございます。いくつかコメントをつけました。

hottahd committed:
- pyproject.tomlにpyvistaを追加
- pyMISO.Dataにwrite_scalar_vtkを追加
- data.i_totalなどとするとdata.conf.i_totalを探しに行く設定にしていたが、confには色々なものが描かれるために、不適と判断。confは探しに行かないようにした。

3D Kelvin Helmholtzと3次元可視化の検討
Fixes #49

hottahd committed:
copilotのコメントに従って改訂

iijimahr reviewed (approved):
ありがとうございます。マージします。
